Set up CI build on Emscripten target

@koute do you know of any existing projects that test every commit on Emscripten? I would have liked to catch #1365 in CI. Is it as simple as just passing a `--target` to Cargo? Does that only let us test cross-compilation or is there a way to run our test suite as well?

I couldn't get asmjs working because of

```console
error[E0463]: can't find crate for `proc_macro`
  --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.4.16/src/lib.rs:53:1
   |
53 | extern crate proc_macro;
   | ^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate
```

but I added a wasm32-unknown-emscripten builder in #1373.

--- a/test_suite/tests/test_de.rs
+++ b/test_suite/tests/test_de.rs
@@ -272,7 +272,7 @@ declare_tests! {
         0f32 => &[Token::F32(0.)],
         0f64 => &[Token::F64(0.)],
     }
-    #[cfg(not(target_arch = "wasm32"))]
+    #[cfg(not(any(target_arch = "asmjs", target_arch = "wasm32")))]
     test_small_int_to_128 {
         1i128 => &[Token::I8(1)],
         1i128 => &[Token::I16(1)],

--- a/test_suite/tests/test_ser.rs
+++ b/test_suite/tests/test_ser.rs
@@ -618,7 +618,7 @@ fn test_enum_skipped() {
     );
 }
 
-#[cfg(not(target_arch = "wasm32"))]
+#[cfg(not(any(target_arch = "asmjs", target_arch = "wasm32")))]
 #[test]
 fn test_integer128() {
     assert_ser_tokens_error(&1i128, &[], "i128 is not supported");

--- a/test_suite/tests/test_value.rs
+++ b/test_suite/tests/test_value.rs
@@ -26,7 +26,7 @@ fn test_u32_to_enum() {
     assert_eq!(E::B, e);
 }
 
-#[cfg(not(target_arch = "wasm32"))]
+#[cfg(not(any(target_arch = "asmjs", target_arch = "wasm32")))]
 #[test]
 fn test_integer128() {
     let de_u128 = IntoDeserializer::<value::Error>::into_deserializer(1u128);

--- a/travis.sh
+++ b/travis.sh
@@ -53,6 +53,7 @@ elif [ -n "${EMSCRIPTEN}" ]; then
     chmod +x ~/.cargo/bin/cargo-web
 
     cd "$DIR/test_suite"
+    cargo web test --target=asmjs-unknown-emscripten --nodejs
     cargo web test --target=wasm32-unknown-emscripten --nodejs
 else
     CHANNEL=nightly

